src/hooks/useENSRegistration.ts

```diff
@@ -0,0 +1,157 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useAccountSettings } from '.';
+import { fetchRegistrationDate } from '@rainbow-me/handlers/ens';
+import {
+  getAvailable,
+  getNameExpires,
+  getRentPrice,
+} from '@rainbow-me/helpers/ens';
+import { Network } from '@rainbow-me/helpers/networkTypes';
+import {
+  convertAmountAndPriceToNativeDisplay,
+  divide,
+  fromWei,
+} from '@rainbow-me/helpers/utilities';
+import { ethereumUtils } from '@rainbow-me/utils';
+
+const WAITING_STATUS = 'waiting';
+const FAILED_STATUS = 'failed';
+const SUCCESS_STATUS = 'success';
+const LOADING_STATUS = 'loading';
+
+const secsInYear = 31536000;
+
+const getRentPricePerYear = (rentPrice: string, duration: number) =>
+  divide(rentPrice, duration);
+
+const formatRentPrice = (
+  rentPrice: string,
+  duration: number,
+  nativeCurrency: any
+) => {
+  const rentPriceInETH = fromWei(rentPrice.toString());
+  const rentPricePerYear = getRentPricePerYear(rentPriceInETH, duration);
+  const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
+    Network.mainnet
+  );
+  const { amount, display } = convertAmountAndPriceToNativeDisplay(
+    rentPriceInETH,
+    nativeAssetPrice,
+    nativeCurrency
+  );
+  const {
+    display: displayPerYear,
+    amount: amountPerYear,
+  } = convertAmountAndPriceToNativeDisplay(
+    rentPricePerYear,
+    nativeAssetPrice,
+    nativeCurrency
+  );
+
+  return {
+    perYear: {
+      amount: amountPerYear,
+      display: displayPerYear,
+    },
+    total: {
+      amount,
+      display,
+    },
+    wei: rentPrice.toString(),
+  };
+};
+
+const formatTime = (timestamp: string) => {
+  const date = new Date(Number(timestamp) * 1000);
+  return `${date.toLocaleDateString()}`;
+};
+
+export default function useENSRegistration({
+  duration,
+  name,
+}: {
+  duration: number;
+  name: string;
+}) {
+  const [registrationData, setRegistrationData] = useState<{
+    available: boolean | null;
+    rentPrice: {
+      total: { amount: string; display: string };
+      perYear: { amount: string; display: string };
+      wei: string;
+    } | null;
+    expirationDate: string | null;
+    registrationDate: string | null;
+  }>({
+    available: null,
+    expirationDate: null,
+    registrationDate: null,
+    rentPrice: null,
+  });
+
+  const { nativeCurrency } = useAccountSettings();
+
+  const nameIsValid = useMemo(() => name.length > 2, [name.length]);
+
+  // status is going to depend if the name is ready and also if is available or not
+  const status = useMemo(() => {
+    if (!nameIsValid) {
+      return WAITING_STATUS;
+    } else if (registrationData.available === null) {
```

Comment:
hmm, this means that the loading state will only be triggered once, and all subsequent loading states won't display.

https://www.loom.com/share/667a31b79ef14543a37e87aa6f06bdf4

can we not use `useQuery` here anymore? otherwise, if we don't care about caching, maybe have an `isLoading` variable which is set at the beginning of the `useEffect` below?